applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_isotropic_damage.cpp

```diff
@@ -220,6 +229,65 @@ void GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::FinalizeMateria
 template <class TConstLawIntegratorType>
 void GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::FinalizeMaterialResponseCauchy(ConstitutiveLaw::Parameters& rValues)
 {
+    // Integrate Stress Damage
+    const Flags& r_constitutive_law_options = rValues.GetOptions();
+
+    // We get the strain vector
+    Vector& r_strain_vector = rValues.GetStrainVector();
+
+    //NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
+    if( r_constitutive_law_options.IsNot( ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN )) {
+        this->CalculateValue(rValues, STRAIN, r_strain_vector);
+    }
+
+    // Elastic Matrix
+    if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) ) {
+        Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();
+        this->CalculateValue(rValues, CONSTITUTIVE_MATRIX, r_constitutive_matrix);
+    }
+
+    // We compute the stress
+    if (r_constitutive_law_options.Is(ConstitutiveLaw::COMPUTE_STRESS)) {
+        // Elastic Matrix
+        Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();
+        this->CalculateValue(rValues, CONSTITUTIVE_MATRIX, r_constitutive_matrix);
+
+        if (r_constitutive_law_options.IsNot(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
+            BaseType::CalculateCauchyGreenStrain( rValues, r_strain_vector);
+        }
+
+        // Converged values
+        double threshold = this->GetThreshold();
+        double damage = this->GetDamage();
+
+        // S0 = C:E
+        array_1d<double, VoigtSize> predictive_stress_vector = prod(r_constitutive_matrix, r_strain_vector);
+
+        // Initialize Plastic Parameters
+        double uniaxial_stress;
+        TConstLawIntegratorType::YieldSurfaceType::CalculateEquivalentStress(predictive_stress_vector, r_strain_vector, uniaxial_stress, rValues);
+
+        const double F = uniaxial_stress - threshold;
```

Comment:
Can you do it in the same way is done in the isotropic plasticity?, with the return value. This reduces the code duplication

Reply:
yeah I know what you mena but it is important to note that the damage does not require to call the integrator previously to the `if (F < 0)` so in this case it is not duplicated i think

Reply:
I say it for the sake of consistency

Reply:
okkey but which method should return the F? the damage integrator do not have the "CalculatePlasticParameters"... :S

Reply:
Certantly, Ok, I will not block because of this. But check the tests

Reply:
In order to update the tests values has to be done in this PR?

Reply:
Yes, because it will not pass Travis